src/java/org/apache/cassandra/db/partitions/AtomicBTreePartition.java

```diff
@@ -135,6 +135,7 @@ protected boolean canHaveShadowedData()
         }
 
         RegularAndStaticColumns columns = update.columns().mergeTo(current.columns);
+        updater.allocated(columns.unsharedHeapSize() - current.columns.unsharedHeapSize());
```

Comment:
I don't think we should track this size, because the columns don't appear in large number in the heap dumps -- I think we manage to reuse the same object though I don't immediately see how we achieve that.

Reply:
removed

Reply:
The `RegularAndStaticColumns` are created during the parsing of the INSERT statement. By default CQLTester used prepared statements. By consequence we only create a single `RegularAndStaticColumns` instance within the stress test. Which does not represent the real life. We should run the stress tests with the use of prepared statement disabled to get a more accurate picture of the impact of the `Columns` objects.

Reply:
After adding` CQLTester.disablePreparedReuseForTest();`
Result before acknowledging the RegularAndStaticColumns is:
`java.lang.AssertionError: Expected heap usage close to 42.305MiB, got 38.662MiB.`
Result after:
`java.lang.AssertionError: Expected heap usage close to 42.302MiB, got 40.212MiB.`
I would suggest to add it and adjust the threshold as  guess the unmetered part is what is missing here, WDYT?

Reply:
I agree with your suggestion.